src/order.rs

```diff
@@ -25,6 +25,10 @@ impl BtcDaiOrder {
         mid_market_rate: Rate,
         spread: Spread,
     ) -> anyhow::Result<BtcDaiOrder> {
+        if base_balance < base_reserved_funds + base_fees {
```

Comment:
I added these checks because if we calculate a minus number the calculation panics. 

There are other solutions for this - instead of using the `-` operator we could also go for the underlying functions that return Options. But I thought avoiding the error is probably the more intuitive way.
Since we use this function on multiple occasions outside I decided that the check has to more in here.

Reply:
Please add a proptest on the function to ensure you solved the panic.

Reply:
Good call! @D4nte you might want to have another look, the guarding logic changed. I wonder if the complexity can be reduced.